cdap-common/bin/functions.sh

```diff
@@ -413,7 +413,20 @@ cdap_set_hbase() {
     1.2-cdh*) __compat="${CDAP_HOME}"/hbase-compat-1.2-cdh5.7.0/lib/* ;; # 5.7 and 5.8 are compatible
     1.2*) __compat="${CDAP_HOME}"/hbase-compat-1.1/lib/* ;; # 1.1 and 1.2 are compatible
     "") die "Unable to determine HBase version! Aborting." ;;
-    *) die "Unknown/Unsupported HBase version found: ${HBASE_VERSION}" ;;
+    *-cdh*)
+      if [[ $(cdap_get_conf "hbase.version.resolution.strategy" "${CDAP_CONF}"/cdap-site.xml auto.strict) == 'auto.latest' ]]; then
+        __compat="${CDAP_HOME}"/hbase-compat-1.2-cdh5.7.0/lib/* # if allowed, return the latest CDH HBase version for unsupported HBase version
+      else
+        die "Unknown/Unsupported HBase version found: ${HBASE_VERSION}"
+      fi
+      ;;
+    *)
+      if [[ $(cdap_get_conf "hbase.version.resolution.strategy" "${CDAP_CONF}"/cdap-site.xml auto.strict) == 'auto.latest' ]]; then
+        __compat="${CDAP_HOME}"/hbase-compat-1.1/lib/* # if allowed, return the latest HBase version for unsupported HBase version
+      else
+        die "Unknown/Unsupported HBase version found: ${HBASE_VERSION}"
+      fi
+      ;;
```

Comment:
Proposal: There is now two places where, when a new compat module is released, that requires updating: adding it to the case statement, and setting it as the "latest" version.
What about explicitly using another variable as the latest version, and setting and using it, so that it is clear that when a new version is added, the _latest version must also be updated_?
```
local readonly __latest_hbase_compat_cdh=hbase-compat-1.2-cdh5.7.0
...
__compat="${CDAP_HOME}"/${__latest_hbase_compat_cdh}/lib/*